src/excalidraw-app/collab/Collab.tsx

```diff
@@ -763,6 +789,37 @@ class Collab extends PureComponent<Props, CollabState> {
     CURSOR_SYNC_TIMEOUT,
   );
 
+  onScrollAndZoomChange = throttle(
+    (payload: { zoom: AppState["zoom"]; scroll: { x: number; y: number } }) => {
+      const appState = this.excalidrawAPI.getAppState();
+
+      const { x: x1, y: y1 } = viewportCoordsToSceneCoords(
+        { clientX: 0, clientY: 0 },
+        {
+          offsetLeft: appState.offsetLeft,
+          offsetTop: appState.offsetTop,
+          scrollX: payload.scroll.x,
+          scrollY: payload.scroll.y,
+          zoom: payload.zoom,
+        },
+      );
+
+      const { x: x2, y: y2 } = viewportCoordsToSceneCoords(
+        { clientX: appState.width, clientY: appState.height },
+        {
+          offsetLeft: appState.offsetLeft,
+          offsetTop: appState.offsetTop,
+          scrollX: payload.scroll.x,
+          scrollY: payload.scroll.y,
+          zoom: payload.zoom,
+        },
+      );
+
+      this.portal.socket &&
+        this.portal.broadcastScrollAndZoom({ bounds: [x1, y1, x2, y2] });
```

Comment:
if nothing else, this whole function should be conditional on `portal.socket` existing, not just the broadcast itself